frontend/controller/service-worker.js

```diff
@@ -306,15 +326,25 @@ const swRpc = (() => {
         reject(event.data)
         cleanup()
       }
+      const oncontrollerchange = (event: Event) => {
+        if (maxControllerChanges > 0) {
+          resolve(fn(maxControllerChanges - 1, ...args))
```

Comment:
Does it make sense to call `resolve` here immediately like this?

Or should we first `await` on the return value of `fn` to make sure `waitUntilSwReady()` finishes, and then call `resolve` with the returned promise?